### PR TITLE
Support pipeline

### LINKED ISSUE #248: Support Pipelines
Hello,

Currently the Pipeline API, as defined [here](https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines) is not supported.

Is there any plan to support it?

Regards

### PR DESCRIPTION
**Pull request by bergholdt**
Implements the pipeline API from https://docs.gitlab.com/ce/api/pipelines.html

- list pipelines
- get a single pipeline
- create a new pipeline
- retry failed builds in a pipeline
- cancel a pipelines builds

closes #248

### CODE REVIEW

File: lib/gitlab/client/pipelines.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    # Create a pipeline.
+    #
+    # @example
+    #   Gitlab.pipeline_create(5, 'master')
+    #
+    # @param  [Integer] project The ID of a project.
+    # @param  [String] ref Reference to commit.
+    # @return [Gitlab::ObjectifiedHash] The pipelines changes.
+    def pipeline_create(project, ref)
```
Comment (asedge) [inline lib/gitlab/client/pipelines.rb]: I believe the pattern has been to do verb_noun for method naming, e.g. `create_pipeline`.

File: lib/gitlab/client/pipelines.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    # Cancels a pipeline.
+    #
+    # @example
+    #   Gitlab.pipeline_cancel(5, 1)
+    #
+    # @param  [Integer] project The ID of a project.
+    # @param  [Integer] id The ID of a pipeline.
+    # @return [Gitlab::ObjectifiedHash] The pipelines changes.
+    def pipeline_cancel(project, id)
```
Comment (asedge) [inline lib/gitlab/client/pipelines.rb]: I believe the pattern has been to do verb_noun for method naming, e.g. `cancel_pipeline`.

File: lib/gitlab/client/pipelines.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    # Retry a pipeline.
+    #
+    # @example
+    #   Gitlab.pipeline_retry(5, 1)
+    #
+    # @param  [Integer] project The ID of a project.
+    # @param  [Integer] id The ID of a pipeline.
+    # @return [Array<Gitlab::ObjectifiedHash>] The pipelines changes.
+    def pipeline_retry(project, id)
```
Comment (asedge) [inline lib/gitlab/client/pipelines.rb]: I believe the pattern has been to do verb_noun for method naming, e.g. `retry_pipeline`.

**Comment by bergholdt** [comment]
@asedge You are right that verb_noun is the preferred pattern for naming.

**Comment by asedge** [comment]
Thank you!

**Pull request merged**